src/components/NewExperiment.vue

```diff
@@ -1699,14 +1702,40 @@ export default Vue.extend({
       }
 
       dialogSelection.then(rowPairsFromDialog => {
-        // parsedRows = [[["name", "a"], ["measurement", 5], ["uncertainty", null]]]
-        parsedRows.forEach((rowPairs, rowIx) => {
-          if (!table.items[rowOffset + rowIx]) {
-            // Create excess rows.
-            table.items.push({ temporaryId: uuidv4() });
-          }
-          [...rowPairs, ...rowPairsFromDialog].forEach(([property, value]) => {
-            Vue.set(table.items[rowOffset + rowIx], property, value);
+        // Transpose the array, so we can make batch requests
+        const columns = zip(...rows);
```

Comment:
This might fail for tens of thousands of rows (max number of arguments given to a function).
Try `unzip(rows)` instead